### PR TITLE
refactor(🔄): use sync queue callback approach to display success toast

### PR DESCRIPTION
**Pull request by jomifepe**
### Description

Alternative approach to https://github.com/prismicio/slice-machine/pull/1436

### How to QA [^1]

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

File: packages/slice-machine/src/features/sync/AutoSyncProvider.tsx [hunk 1]
```diff
@@ -72,79 +73,85 @@ export const AutoSyncProvider: FC<PropsWithChildren> = (props) => {
...
+              // know that they need to login again.
+              // This can easily happen if the the token expires when the user is
+              // offline.
+              return;
+            }
+
+            // If the sync failed, we want to display the error message with
+            // the retry button.
+            reject?.(error);
+            throw error;
```
Comment (dani-mp) [inline packages/slice-machine/src/features/sync/AutoSyncProvider.tsx]: Two things here:
- above if `isUnauthenticatedError(error)`, we're not resolving/rejecting the promise, what should we do here?
- here, i don't think we want to both reject and throw, it's like bubbling the error twice

File: packages/slice-machine/src/features/sync/AutoSyncProvider.tsx [hunk 2]
```diff
@@ -42,14 +42,15 @@ export type AutoSyncStatus =
...
-  syncChanges: (args?: SyncChangesArgs) => void;
+  syncChanges: (args?: SyncChangesArgs) => void | Promise<void>;
 };
 
 type SyncChangesArgs = {
   environment?: Environment;
   loggedIn?: boolean;
   changedCustomTypes?: ChangedCustomType[];
   changedSlices?: ChangedSlice[];
+  callback?: () => void;
```
Comment (dani-mp) [inline packages/slice-machine/src/features/sync/AutoSyncProvider.tsx]: we don't need this, right?

File: packages/slice-machine/src/features/sync/AutoSyncProvider.tsx [hunk 3]
```diff
@@ -42,14 +42,15 @@ export type AutoSyncStatus =
 
 type AutoSyncContext = {
   autoSyncStatus: AutoSyncStatus;
-  syncChanges: (args?: SyncChangesArgs) => void;
+  syncChanges: (args?: SyncChangesArgs) => void | Promise<void>;
```
Comment (dani-mp) [inline packages/slice-machine/src/features/sync/AutoSyncProvider.tsx]: why not?
```suggestion
  syncChanges: (args?: SyncChangesArgs) => Promise<void>;
```

Comment (jomifepe) [reply packages/slice-machine/src/features/sync/AutoSyncProvider.tsx]: Forgot to remove it 😅

File: packages/slice-machine/src/features/sync/AutoSyncProvider.tsx (hunk 1 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/sync/AutoSyncProvider.tsx]: Yeah makes sense, fixed!

**Pull request merged**